libs/s25main/mapGenerator/README.md

```diff
@@ -1,41 +0,0 @@
-# Random Map Generation
```

Comment:
Feel free to keep that. If it is up to date t might help future developers :)

Reply:
Fair enough. For now I removed it because it doesn't seem to contain information which are worth the additional effort for maintaining it. I might add it later after refactoring the classes (and moving some stuff to free functions as suggested) to explain how to add new "random map types" or tweak the generation process. 

Let me know if you find any commits unreasonable, your comments in between helped quite a bit (if you have time to check ofc). The video on free functions was quite fun to watch.